kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/AnalysisExecutor.kt

```diff
@@ -0,0 +1,48 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ex.ProjectManagerEx
+import org.jetbrains.research.ml.kotlinAnalysis.util.getSubdirectories
+import java.nio.file.Path
+
+/**
+ * Abstract class for analysis executor which provides interface for execution analysis
+ * for each project in given dataset.
+ */
+abstract class AnalysisExecutor {
+
+    /** Executes the analysis of the given [project][Project]. */
+    abstract fun analyse(project: Project)
+
+    /**
+     * Finishes the task execution. Should contain close functions of all writers and readers
+     * and called after all project analysis execution finished.
+     */
+    abstract fun finish()
+
+    /** Execute analysis for all projects in [given directory][projectsDir]. */
+    fun execute(projectsDir: Path) {
+        try {
+            getSubdirectories(projectsDir).forEach { projectPath ->
+                ApplicationManager.getApplication().runReadAction {
+                    println("Opening project $projectPath")
+                    ProjectUtil.openOrImport(projectPath, null, true).let { project ->
```

Comment:
Should we send a function into `execute` function as an argument that can open or set up a project? I mean for several tasks we should use not only `openOrImport` function. I mean something like this: setupProject: (String) -> Project
Moreover, you should use 
```
ApplicationManager.getApplication().invokeAndWait 
```
to run this function to finish all indexing proceses